src/main/resources/frontend/src/network/ServerCommunicator.ts

```diff
@@ -0,0 +1,70 @@
+import { useAuthStore } from '@/stores/auth'
+import { ServerError } from '@/network/ServerError'
+import { useAppConfigStore } from '@/stores/appConfig'
+
+export const ServerCommunicator = {
+  getRequest: getRequest,
+  postRequest: postRequest,
+  patchRequest: patchRequest,
+  doUnprocessedRequest: doUnprocessedRequest
+}
+
+async function getRequest<T>(endpoint: string, expectResponse?: boolean): Promise<T> {
+  return await doRequest<T>("GET", endpoint, null,expectResponse ?? true)
+}
+
+async function postRequest<T>(endpoint: string, bodyObject?: Object, expectResponse?: boolean): Promise<T> {
+  return await doRequest<T>("POST", endpoint, bodyObject,expectResponse ?? true)
+}
+
+async function patchRequest<T>(endpoint: string, bodyObject?: Object, expectResponse?: boolean): Promise<T> {
+  return await doRequest<T>("PATCH", endpoint, bodyObject,expectResponse ?? true)
+}
+
+async function doRequest<T>(method: string,
+                            endpoint: string,
+                            bodyObject?: Object | null,
+                            expectResponse?: boolean): Promise<T> {
+  const response = await doUnprocessedRequest(method, endpoint, bodyObject)
+
+  if (!expectResponse) {
+    return null as T
```

Comment:
This is strange. Why are we typecasting `null` to `T` when it never will be?

We can express this relationship more clearly with TypeScript function overload signatures. Note that when signature overloads are provided in TS, the last function declaration is interpreted as the "implementation signature." The last signature must be compatible with all the previous "call signatures," and it is not exposed for public use.

```ts
function doRequest<T>(
    method: string,
    endpoint: string,
    bodyObject: Object | null,
    expectResponse: false,
): Promise<null>
function doRequest<T>(
    method: string,
    endpoint: string,
    bodyObject: Object | null,
    expectResponse: true,
): Promise<T>
function doRequest<T>(
    method: string,
    endpoint: string,
    bodyObject?: Object | null,
    expectResponse?: true,
): Promise<T | null>
function doRequest<T>(
    method: string,
    endpoint: string,
    bodyObject?: Object | null,
    expectResponse?: boolean,
): Promise<T | null> { /* implementation */ }
```

Reply:
Alternatively, the main point of these changes could also be achieved with explicit TSDocumentation:
```ts

/**
 * Conveniently submits an HTTP Request and returns the intepreted result.
 * 
 * Only interprets and returns the response body when such a body is expected.
 * The response body is parsed into JSON, but it is not validated or guaranteed
 * to be the expected `T` type. The safest way to use this method is to set `<T=unknown>`.
 * 
 * @param method HTTP Method to submit
 * @param endpoint URL Endpoint to hit
 * @param bodyObject Optional. An object containing JSON body data to send with the Request
 * @param expectResponse Default: `false`. The response body will only be parsed when set to `true`.
 * @returns A {@linkcode Promise} resolving to the generic T (unsafely cast) when `expectResponse=true`, 
 *          and `null` all the rest of the time.
 */
function doRequest<T>(
    method: string,
    endpoint: string,
    bodyObject?: Object | null,
    expectResponse?: boolean,
): Promise<T | null>
```

Reply:
Additionally, the two approaches could be combined to result in the most explicitly documented code possible.

Reply:
I'll go through and document the new code. The whole `if (expectResponse) { return null as T }` shenanigan was the only way I could figure out how to get it to return null when no response was wanted. Ideally, if you didn't want a response, you would pass `null` or `void` in as the type. But that doesn't work in typescript since types don't really exist at runtime.

Reply:
I'm a bit confused about what you are proposing with the overloads. How do the different signatures allow the user to get a null back?